root/static/scripts/common/immutable-entities.js

```diff
@@ -37,7 +37,7 @@ const reduceArtistCredit = ac => ac.names.reduce(reduceName, '');
 const isComplexArtistCredit = function (ac) {
   const firstName = ac.names.get(0);
   if (firstName && hasArtist(firstName)) {
-     return firstName.artist.name !== reduceArtistCredit(ac);
+     return firstName.artist.name !== reduceArtistCredit(ac) || !nonEmpty(firstName.name);
```

Comment:
Yeah, I think this makes sense. My only recommendation would be to take advantage of short-circuit evaluation and put the `!nonEmpty(firstName.name)` check before the first one, since `reduceArtistCredit` is more expensive.